src/SAWScript/Crucible/LLVM/ResolveSetupValue.hs

```diff
@@ -458,6 +476,7 @@ resolveSetupVal cc mem env tyenv nameEnv val =
       | Just ptr <- Map.lookup i env -> return (Crucible.ptrToPtrVal ptr)
       | otherwise -> fail ("resolveSetupVal: Unresolved prestate variable:" ++ show i)
     SetupTerm tm -> resolveTypedTerm cc tm
+    SetupCast () v _ -> resolveSetupVal cc mem env tyenv nameEnv v
```

Comment:
If I understand correctly, it's OK to simply discard the `Type` argument here because casted things will always resolve to a pointer (i.e., an `LLVMValInt`), correct? If so, it would be worth adding a brief comment here to that effect.

Reply:
Yes, casted values should always be pointers.